sui-execution/next-vm/sui-adapter/src/execution_engine.rs

```diff
@@ -81,6 +81,7 @@ mod checked {
         certificate_deny_set: &HashSet<TransactionDigest>,
     ) -> (
         InnerTemporaryStore,
+        SuiGasStatus,
         TransactionEffects,
         Result<Mode::ExecutionResults, ExecutionError>,
     ) {
```

Comment:
this is not about this PR specifically but this PR made me think about it and I figured I'd ask more broadly (@lxfind @bmwill @amnn @todd-mystenlabs @tzakian @cgswords).
Here the size of this return value is pretty massive and it gets propagated even more. Implication is that we are doing a bunch of big copies across functions. 
I looked at some of the usages and it's not clear to me that it is necessarily good or bad, but certainly we use those values multiple times and by reference often.
So I am wondering if it is good to return such a big chunk of memory or if we would be better off boxing and effectively make those a reference.
Not sure, I figured I'd ask

Reply:
I would have to look at the disassembly to be sure, but I believe Rust supports RVO and copy-elision, like C++, so this may not be being copied over and over again.

Reply:
yeah I understand the compiler may optimize it but it depends a lot on the code, where the objects are created and where they are used, so do not know...
but anyway thanks for the reply and it did not seem anybody cared that much so we are good